erdos/op.py

```diff
@@ -86,14 +92,22 @@ def spin(self):
     def log_event(self, processing_time, timestamp, log_message=None):
         pass
 
+    def record_streams(self, stream_uid, msg):
```

Comment:
IMHO, recording messages of a stream should not happen in the operator. Recording a stream is an activity that is strictly associated with a stream. If we follow the code path, we now invoke `record_streams` from the stream, and we even pass the message for the recording to happen. I would move this logic into the data streams. As a result, we would also handle for free cases such as recording only once data streams that multiple receivers.